src/main/webapp/MyMarker.js

```diff
@@ -12,36 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-class Marker {
+class MyMarker {
 
   /** The google.maps.Marker object visible on the map*/
-  marker;
+  googleMarker;
 
+  /** Information window associated with this marker */
+  myInfoWindow;
+
+  /**
+   * @param {google.maps.LatLng} coords the coordinates for the marker
+   */
   constructor(coords) {
-    this.marker = new google.maps.Marker(
+    this.googleMarker = new google.maps.Marker(
       {
         position: coords,
-        map: myMap.map
+        map: myMap.googleMap
       });
+    this.myInfoWindow = new MyInfoWindow(this);
     this.setMarkerListeners();
+    this.myInfoWindow.open();
   }
 
   /**
    * Sets marker-triggered events
    */
   setMarkerListeners() {
     let _self = this;
-    _self.marker.addListener('click', () => {
+    _self.googleMarker.addListener('dblclick', () => {
```

Comment:
When you use the short-hand notation of functions like you did here, `this` would still point to the outer-scoped object. You only need `_self` if you use the `function` keyword, which creates a new scope. Please remove `_self` from here. You don't need to store `this` in `_self` here.

Reply:
Right, I think I encountered the problem with `this` being out of scope in ChapMap so I used `_self` to fix that but I didn't realize arrow functions don't have that problem. Now I removed `_self` from my code and changed the Chapmap function that was causing problems into an arrow function.